src/skmultiflow/data/data_generator.py

```diff
@@ -0,0 +1,74 @@
+import pandas as pd
+import numpy as np
+from skmultiflow.data.base_generator import BaseGenerator
+
+# TODO : doc for all methods
+
+class DataGenerator(BaseGenerator):
+    # TODO : inherit from pandas.DataFrame should make things easier
+    """ DataGenerator
+
+    A generator constructed from the entries of a static dataset (numpy array or pandas
```

Comment:
I like the idea of having a single method for both **file** and **raw data** sources. However, it would be nice to make clear in the example section how both functionalities remain. This is because a lot of users rely on these methods. Optionally we could keep `DataStream` and `FileStreams` as wrappers with a deprecation warning.

Reply:
Unless the users are well-involved in the development process, I would second deprecation or clear documentation on how to switch.

Reply:
File sources are already tackled by streamz via `Source.from_textfile`

I think I can add a deprecation warning

Reply:
@jacobmontiel for this case I don't see how a deprecation warning could help, taking the renaming into account

If users had pieces of code importing FileStream, they could not import it anyway because following the applied renaming it would be rename to FileGenerator ...

Reply:
I mean to have a wrapper class `FileStream` (same name as before) that internally calls the new method. If the user calls `FileStream` then the warning is raised.